src/EventHub.Domain/Organizations/Plans/PlanFeatureManager.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Threading.Tasks;
+using EventHub.Events;
+using EventHub.Events.Registrations;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Domain.Repositories;
+
+namespace EventHub.Organizations.Plans;
+
+public class PlanFeatureManager : ITransientDependency
+{
+    private readonly IPlanInfoDefinitionStore _planInfoDefinitionStore;
+    private readonly IRepository<Organization, Guid> _organizationRepository;
+    private readonly IRepository<Event, Guid> _eventRepository;
+    private readonly IEventRegistrationRepository _eventRegistrationRepository;
+
+    public PlanFeatureManager(
+        IPlanInfoDefinitionStore planInfoDefinitionStore, 
+        IRepository<Organization, Guid> organizationRepository, 
+        IRepository<Event, Guid> eventRepository, 
+        IEventRegistrationRepository eventRegistrationRepository)
+    {
+        _planInfoDefinitionStore = planInfoDefinitionStore;
+        _organizationRepository = organizationRepository;
+        _eventRepository = eventRepository;
+        _eventRegistrationRepository = eventRegistrationRepository;
+    }
+    
+    public virtual async Task<bool> CanCreateNewEvent(Guid organizationId)
+    {
+        var currentPlanOfOrganization = await GetCurrentPlanByOrganizationIdAsync(organizationId);
+        
+        var totalEventCountByOrganization = await _eventRepository.CountAsync(x => x.OrganizationId == organizationId && x.CreationTime.Year < DateTime.Now.Year + 1 && x.CreationTime.Year > DateTime.Now.Year - 1);
```

Comment:
```suggestion
        var totalEventCountByOrganizationInThisYear = await _eventRepository.CountAsync(x => x.OrganizationId == organizationId && x.CreationTime.Year == DateTime.Now.Year);
```

Why not this? Also, are you sure that `x.CreationTime.Year` is properly translated to SQL. I curious if it is executed in client side. In addition, shouldn't we use IClock.Now instead of DateTime.Now?

Reply:
Entity Framework Core now blocks such client evaluation and raises a runtime exception. For this reason, it is now necessary to specify the client evaluation explicitly, if desired.

**References:**
1. https://docs.microsoft.com/en-us/ef/core/querying/client-eval#unsupported-client-evaluation
2. https://docs.microsoft.com/en-us/ef/core/querying/client-eval#previous-versions